docs/concepts/libp2p.md

```diff
@@ -33,43 +33,43 @@ Libp2p works with a lot of different addressing schemes in a consistent way. A m
 
 The technology used to move your data from one machine to another. Transports are defined in terms of two core operations, _listening_ and _dialing_. Listening means that you can accept incoming connections from other peers. _Dialing_ is the process of opening an outgoing connection to a listening peer. One of Libp2p's core requirements is to be _transport agnostic_, meaning the decision of what transport protocol to use is up to an application's developer (who may decide to support many different _transports_ at the same time).
 
-#### [Security](https://docs.libp2p.io/introduction/#secure-communication)
+#### [Security](https://docs.libp2p.io/introduction/what-is-libp2p/#security)
 
 Libp2p supports upgrading a transport connection into a securely encrypted channel. You can then trust the identity of the peer you're communicating with and that no third-party can read the conversation or alter it in-flight. The current default is [TLS 1.3](https://www.ietf.org/blog/tls13/) as of IPFS 0.7. The previous default of [SECIO](https://docs.libp2p.io/concepts/secure-comms/) is now deprecated and disabled by default (see [this blog post](https://blog.ipfs.tech/2020-08-07-deprecating-secio/) for more information).
 
-#### [Peer identity](https://docs.libp2p.io/reference/glossary/#peerid)
+#### [Peer identity](https://docs.libp2p.io/concepts/peers/#peer-id)
 
 A Peer Identity ([often written _PeerId_](https://docs.libp2p.io/reference/glossary/#peerid)) is a unique reference to a specific peer on the peer-to-peer network. Each Libp2p peer has a private key, which it keeps secret from all other peers, and a corresponding public key, which is shared with other peers. The PeerId is a [cryptographic hash](https://en.wikipedia.org/wiki/Cryptographic_hash_function) of a peer's public key. PeerIds are encoded using the [multihash](https://docs.libp2p.io/reference/glossary/#multihash) format.
 
 #### [Peer routing](https://docs.libp2p.io/introduction/#peer-routing)
 
 Peer Routing is the process of discovering peer addresses by using the knowledge of other peers. In a peer routing system, a peer can either give us the address we need if they have it or else send our inquiry to another peer who's more likely to have the answer. Peer Routing in Libp2p uses a [distributed hash table](https://docs.libp2p.io/reference/glossary/#dht) to iteratively route requests closer to the desired PeerId using the [Kademlia](https://en.wikipedia.org/wiki/Kademlia) routing algorithm.
 
-#### [Content discovery](https://docs.libp2p.io/introduction/#content-discovery)
+#### [Content discovery](https://docs.libp2p.io/introduction/what-is-libp2p/#content-discovery)
```

Comment:
concepts/introduction/overview/#content-discovery

Reply:
So, this link https://docs.libp2p.io/concepts/introduction/overview/#content-discovery doesn't appear to work properly.

This link https://docs.libp2p.io/concepts/introduction/overview/?#content-discovery does. Not sure why the "?" is there. Thoughts @DannyS03